### PR TITLE
apps wc: Allow user to list and run top cluster wide

### LINKED ISSUE #850: Allow users to list pods and get metrics cluster-wide
**Is your feature request related to a problem? Please describe.**
To allow users to easily determine resource allocation on nodes they need to be able to list pods cluster-wide in order to run `k describe node`.
Listing pods should not reveal any information that a user does not already have access to via Kibana and Grafana.

Being able to get metrics cluster-wide enables users to run `k top nodes` to quickly be able to determine resource usage exposed via the metrics-server.

**Describe the solution you'd like**
Users should be assigned the following cluster-wide permissions
```
- apiGroups:
  - metrics.k8s.io
  resources:
  - pods
  - nodes
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - ""
  resources:
  - pods
  verbs:
  - list
```

Note that I'm not 100% sure if they need `list` and `watch` for the metrics or if it is enough with `get`.

**Describe alternatives you've considered**
Not giving users more privileges.

**Definition of done:**
A user  can run `k describe nodes` and see pods and allocated resources, and they can run `k top nodes` and see usage information.

### PR DESCRIPTION
**Pull request by OlleLarsson**
**What this PR does / why we need it**:
Increases the privileges for user admins to be able to list pods cluster wide, which is needed to be able to see the allocated resources when running the `k describe pod` command.
Also gives access to metrics cluster wide to be able to run `k top nodes`.

**Which issue this PR fixes**:
fixes #850 

**Public facing documentation PR** *(if applicable)*

**Special notes for reviewer**:
Added `olle.larsson@elastisys.com` as an admin user:

```
❯ k auth can-i --as=olle.larsson@elastisys.com list pods --all-namespaces 
yes
```

```
❯ k describe nodes --as=olle.larsson@elastisys.com
...
Non-terminated Pods:          (22 in total)
  Namespace                   Name                                                          CPU Requests  CPU Limits   Memory Requests   Memory Limits    Age
  ---------                   ----                                                          ------------  ----------   ---------------   -------------    ---
  cert-manager                cert-manager-webhook-86587486bd-gvdf5                         25m (0%)      250m (6%)    25Mi (0%)         250Mi (3%)       34d
  falco                       falco-exporter-qfkdf                                          30m (0%)      30m (0%)     20Mi (0%)         20Mi (0%)        34d
  falco                       falco-hpnm8                                                   100m (2%)     200m (5%)    512Mi (6%)        1Gi (13%)        34d
  fluentd                     fluentd-fluentd-elasticsearch-fdlrn                           200m (5%)     500m (12%)   300Mi (3%)        572Mi (7%)       34d
  gatekeeper-system           gatekeeper-controller-manager-795447c446-lbdpg                100m (2%)     1 (25%)      256Mi (3%)        512Mi (6%)       28d
  gatekeeper-system           gatekeeper-controller-manager-795447c446-rp6vn                100m (2%)     1 (25%)      256Mi (3%)        512Mi (6%)       28d
  ingress-nginx               ingress-nginx-controller-jgg48                                100m (2%)     200m (5%)    263Mi (3%)        512Mi (6%)       18d
  kube-system                 calico-node-f9tbf                                             150m (3%)     300m (7%)    64M (0%)          500M (6%)        80d
  kube-system                 coredns-8474476ff8-p5q7m                                      100m (2%)     0 (0%)       70Mi (0%)         170Mi (2%)       80d
  kube-system                 kube-proxy-dk9gp                                              0 (0%)        0 (0%)       0 (0%)            0 (0%)           80d
  kube-system                 nginx-proxy-olle-wc-worker-2                                  25m (0%)      0 (0%)       32M (0%)          0 (0%)           80d
  kube-system                 node-local-dns-rcqwr                                          25m (0%)      0 (0%)       40Mi (0%)         0 (0%)           34d
  monitoring                  kube-prometheus-stack-prometheus-node-exporter-nnwqq          25m (0%)      50m (1%)     50Mi (0%)         100Mi (1%)       34d
  monitoring                  starboard-operator-6b9956968d-8z6kd                           10m (0%)      20m (0%)     128Mi (1%)        172Mi (2%)       34d
  monitoring                  vulnerability-exporter-65bc4c6dcf-rjnw2                       40m (1%)      80m (2%)     79Mi (1%)         158Mi (2%)       34d
  postgres-system             iam-main-0                                                    1050m (26%)   1500m (38%)  1173741824 (14%)  1298576Ki (16%)  4d22h
  rook-ceph                   csi-rbdplugin-ftscs                                           15m (0%)      0 (0%)       35Mi (0%)         0 (0%)           80d
  rook-ceph                   csi-rbdplugin-provisioner-747b4fcbd4-tpmp6                    30m (0%)      0 (0%)       140Mi (1%)        0 (0%)           80d
  rook-ceph                   rook-ceph-crashcollector-olle-wc-worker-2-65cc9bc6bb-8vgwt    1m (0%)       0 (0%)       15Mi (0%)         0 (0%)           80d
  rook-ceph                   rook-ceph-mgr-a-765d9b9688-6wmnh                              100m (2%)     0 (0%)       500Mi (6%)        0 (0%)           80d
  rook-ceph                   rook-ceph-mon-a-86cd9d746f-dvbmx                              100m (2%)     0 (0%)       1Gi (13%)         0 (0%)           80d
  rook-ceph                   rook-ceph-osd-2-7bb5fd4f46-874hw                              500m (12%)    0 (0%)       1Gi (13%)         0 (0%)           80d
Allocated resources:
  (Total limits may be over 100 percent, i.e., overcommitted.)
  Resource           Requests          Limits
  --------           --------          ------
  cpu                2826m (72%)       5130m (131%)
  memory             6236846336 (78%)  6026142976 (75%)
  ephemeral-storage  0 (0%)            0 (0%)
  hugepages-1Gi      0 (0%)            0 (0%)
  hugepages-2Mi      0 (0%)            0 (0%)
```

```
❯ k top nodes --as=olle.larsson@elastisys.com --use-protocol-buffers 
NAME               CPU(cores)   CPU%   MEMORY(bytes)   MEMORY%   
olle-wc-master-0   326m         18%    2795Mi          84%       
olle-wc-worker-0   484m         12%    4024Mi          52%       
olle-wc-worker-1   598m         15%    4967Mi          65%       
olle-wc-worker-2   333m         8%     5158Mi          67%   
```

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [ ] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

**Pipeline config** *(if applicable)*
If you change some config options (e.g. add/rename variable or change the default value) you may need to update the config used by the pipeline in `pipeline/config`.

### CODE REVIEW

**Comment by Pavan-Gunda** [comment]
LGTM!!

**Review by Pavan-Gunda — APPROVED** [review]

File: helmfile/charts/user-rbac/templates/clusterroles/user-view.yaml [hunk 1]
```diff
@@ -6,3 +6,9 @@ rules:
 - apiGroups: [""]
   resources: ["nodes","namespaces","persistentvolumes"]
   verbs: ["get", "watch", "list"]
+- apiGroups: ["metrics.k8s.io"]
+  resources: ["pods","nodes"]
+  verbs: ["list"]
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["list"]
```
Comment (viktor-f) [inline helmfile/charts/user-rbac/templates/clusterroles/user-view.yaml:14]: Based on our discussion earlier that `list` basically gets them the same privilege as `get`, should we maybe give them `get` as well? Maybe even `watch`?

Comment (OlleLarsson) [reply helmfile/charts/user-rbac/templates/clusterroles/user-view.yaml:14]: I'm okay with that.

File: WIP-CHANGELOG.md [hunk 2]
```diff
@@ -37,6 +37,8 @@
 ### Fixed
 
 ### Added
+- RBAC for admin user so that they now can list pods cluster wide and run the `kubectl top`.
+  - The no longer needed rolebinding and clusterrole `metrics` has been removed.
```
Comment (cristiklein) [inline WIP-CHANGELOG.md:41]:
```suggestion
- The no longer needed rolebinding and clusterrole `metrics` has been removed.
```

File: helmfile/charts/user-rbac/templates/clusterroles/user-view.yaml (hunk 1 above)
Comment (cristiklein) [reply helmfile/charts/user-rbac/templates/clusterroles/user-view.yaml:14]: :+1:

**Review by cristiklein — APPROVED** [review]
Short and easy to review. Nice! :smile:

**Pull request merged**